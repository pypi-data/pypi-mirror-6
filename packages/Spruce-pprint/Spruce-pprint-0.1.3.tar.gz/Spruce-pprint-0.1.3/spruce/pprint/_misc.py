"""Miscellaneous tools."""

__copyright__ = "Copyright (C) 2014 Ivan D Vasin"
__docformat__ = "restructuredtext"


def indented(string, level=1, size=4):
# TODO: (Python 3)
#def indented(string, level=1, *, size=4):
    """Indent a string.

    :param int level:
        The number of indents to apply.

    :param int size:
        The size of each indent.

    :rtype: :obj:`str`

    :raise ValueError:
        Raised if *level* or *size* is negative.

    """
    if level < 0:
        raise ValueError('level must be nonnegative')
    if size < 0:
        raise ValueError('size must be nonnegative')
    if level == 0 or size == 0:
        return string[:]
    return '\n'.join(' ' * size * level + line
                     for line in string.split('\n'))


def split_blocks(string, joinlines=True):
    """Split a string into blocks.

    If *joinlines* is true, then the lines within each block are joined with
    :func:`strip_and_cleanjoin`.

    :param str string:
        A string.

    :param bool joinlines:
        Whether to join the lines within each block.

    :rtype: [:obj:`str`]

    """
    blocks = []
    lines = string.split('\n')
    joinfunc = strip_and_cleanjoin if joinlines else '\n'.join
    blocklines = []
    for line in lines:
        line = line.strip()
        if line:
            blocklines.append(line)
        else:
            if blocklines:
                blocks.append(joinfunc(*blocklines))
                blocklines = []
    if blocklines:
        blocks.append(joinfunc(*blocklines))
    return blocks


def strip_and_cleanjoin(*strings):
    """
    Strip some strings and join them with numbers of spaces that respect
    punctuation.

    .. note:: **TODOC:**
        example

    :param strings:
        Strings.
    :type strings: ~[:obj:`str`]

    :rtype: :obj:`str`

    """
    strings_new = []
    punctuation_by_spaces = {(0, 0): ['--', '---'],
                             (0, 1): [',', ';', ':'],
                             (0, 2): ['.', '!', '?'],
                             (1, 1): ['=', '+', '-', '*', '/', '&', '|', '<',
                                      '>', '~', '\\'],
                             }
    prevstring_endspaces = 0
    for i, string in enumerate(strings):
        string = string.strip()

        # prepend spaces if previous string didn't have enough
        if i == 0:
            pass
        else:
            for startspaces, endspaces in sorted(punctuation_by_spaces.keys()):
                if prevstring_endspaces < startspaces \
                       and any(string.startswith(mark)
                               for mark
                               in punctuation_by_spaces[(startspaces,
                                                         endspaces)]):
                    string = ' ' * (startspaces - prevstring_endspaces) \
                             + string

        # append spaces if string is terminated by a punctuation mark that
        #     demands them
        if i == len(strings) - 1:
            prevstring_endspaces = 0
        else:
            for startspaces, endspaces in sorted(punctuation_by_spaces.keys()):
                if any(string.endswith(mark)
                       for mark in punctuation_by_spaces[(startspaces,
                                                          endspaces)]):
                    prevstring_endspaces = endspaces
                    string += ' ' * endspaces

        strings_new.append(string)
    return ''.join(strings_new)


def stripped_lines(string, chars=' \t', tabsize=8):
    """
    Strip the greatest amount of leading characters that is present in all
    lines of a string.

    In other words, move the left margin as close as possible to the
    non-*chars* content of *string*.

    For example, if *chars* contains the space character, then this
    *string* ::

        '''
            Fancy Heading

            data:
                function: foo
                widgets: 42
        '''

    will be returned like this::

        '''
        Fancy Heading

        data:
            function: foo
            widgets: 42
        '''

    .. note:: **TODO:**
        implement this:
            Tab characters are interpreted as having a size of *tabsize*.
            If the number of characters to strip is not a multiple of
            *tabsize*, then tabs that span the new margin are converted to
            spaces and stripped accordingly.

    :param str string:
        A string to be stripped.

    :param str chars:
        A string of characters to strip.

    :param int tabsize:
        The size of a tab character.

    :rtype: :obj:`str`

    """
    lines = string.split('\n')
    strip_count = None
    def chars_count(line):
        for i, char in enumerate(line):
            if char not in chars:
                return i
    for line in lines:
        count = chars_count(line)
        if strip_count is None:
            strip_count = count
        else:
            strip_count = min(strip_count, count)
        if strip_count == 0:
            return lines[:]
    return '\n'.join(line[strip_count:] for line in lines)
