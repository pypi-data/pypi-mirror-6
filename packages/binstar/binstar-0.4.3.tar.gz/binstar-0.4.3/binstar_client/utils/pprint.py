'''
Created on Aug 8, 2013

@author: sean
'''
from dateutil.parser import parse as parse_date
import logging
log = logging.getLogger('binstar.pprint')

def pprint_orgs(orgs):
    log.info('Organizations:')
    for org in orgs:
        log.info('   + %(login)25s' % org)

fmt_access = '     %(full_name)25s | %(access)-12s | %(package_types)-15s | %(summary)s'
fmt_no_access = '     %(full_name)25s | %(package_types)-15s | %(summary)s'

def pprint_package_header(access=True):
    package_header = {'full_name':'Name', 'access': 'Access', 'package_types':'Package Types', 'summary':'Summary'}
    fmt = fmt_access if access else fmt_no_access
    log.info(fmt % package_header)

def pprint_package(package, access=True, full_name=True):
    package = package.copy()
    if not full_name:
        package['full_name'] = package['name']
    package['access'] = 'published' if package.get('published') else 'public' if package['public'] else 'private'
    if package.get('package_types'):
        package['package_types'] = ', '.join(package['package_types'])
    fmt = fmt_access if access else fmt_no_access
    log.info(fmt % package)
        
def pprint_packages(packages, access=True, full_name=True):
    if packages:
        log.info('Packages:')
    else:
        log.info('No packages found')
    
    fmt = fmt_access if access else fmt_no_access
    pprint_package_header(access)
    
    package_header = {'full_name':'-' * 25, 'access': '-' * 12, 'package_types':'-' * 15, 'summary':'-' * 20}
    
    log.info(fmt % package_header)
    for package in packages:
        pprint_package(package, access, full_name)

def pprint_user(user):
    user = user.copy()
    log.info('Username: %s', user.pop('login'))
    log.info('Member since: %s', parse_date(user.pop('created_at')).ctime())

    for key_value in user.items():
        log.info('  +%s: %s' % key_value)

def pprint_collections(collections):
    if collections:
        log.info('Collections:')
    for collection in collections:
        collection['permission'] = 'public' if collection['public'] else 'private'
        log.info('   + %(name)25s: [%(permission)s] %(description)s' % collection)

