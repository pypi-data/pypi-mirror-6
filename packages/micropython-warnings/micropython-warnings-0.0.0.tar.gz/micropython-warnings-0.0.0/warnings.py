# MicroPython doesn't support warnings
