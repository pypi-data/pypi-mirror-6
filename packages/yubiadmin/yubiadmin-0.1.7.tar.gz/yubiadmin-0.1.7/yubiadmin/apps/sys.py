# Copyright (c) 2013 Yubico AB
# All rights reserved.
#
#   Redistribution and use in source and binary forms, with or
#   without modification, are permitted provided that the following
#   conditions are met:
#
#    1. Redistributions of source code must retain the above copyright
#       notice, this list of conditions and the following disclaimer.
#    2. Redistributions in binary form must reproduce the above
#       copyright notice, this list of conditions and the following
#       disclaimer in the documentation and/or other materials provided
#       with the distribution.
#
# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
# POSSIBILITY OF SUCH DAMAGE.

import os
import subprocess
from webob import Response
from threading import Timer
from yubiadmin.util.app import App, render
from yubiadmin.util.system import run
from yubiadmin.apps.dashboard import panel

__all__ = [
    'app'
]


UPGRADE_LOG = "/var/tmp/yubix-upgrade"


def get_updates():
    s, o = run("apt-get upgrade -s | awk -F'[][() ]+' '/^Inst/{print $2}'")
    packages = o.splitlines()
    return packages


def needs_restart():
    return os.path.isfile('/var/run/reboot-required')


def reboot():
    run('reboot')


class Updater(object):
    def __init__(self):
        self.proc = subprocess.Popen('DEBIAN_FRONTEND=noninteractive '
                                     'apt-get -y dist-upgrade -o '
                                     'Dpkg::Options::="--force-confdef" -o '
                                     'Dpkg::Options::="--force-confold" | '
                                     'tee %s' % UPGRADE_LOG,
                                     stdout=subprocess.PIPE, shell=True)

    def __iter__(self):
        yield """
        <script type="text/javascript">
        function reload() {
            window.location.replace('/sys');
        }
        window.onload = function() {
            setTimeout(reload, 10000);
        }
        </script>
        <strong>Performing update, this may take a while...</strong><br/>
        <pre>
        """

        while True:
            line = self.proc.stdout.readline()
            if line:
                yield line
            else:
                yield '</pre><br /><strong>Update complete!</strong>'
                yield '<script type="text/javascript">reload();</script>'
                break


class SystemApp(App):
    """
    YubiX System
    """
    sections = ['general']
    priority = 30

    @property
    def disabled(self):
        #return not os.path.isdir('/usr/share/yubix')
        return False

    @property
    def hidden(self):
        return self.disabled

    @property
    def dash_panels(self):
        if needs_restart():
            yield panel('System', 'System restart required', level='danger')

        updates = len(get_updates())
        if updates > 0:
            yield panel(
                'System',
                'There are <strong>%d</strong> updates available' % updates,
                '/%s/general' % self.name,
                'info'
            )

        _, time = run('date "+%a, %d %b %Y %H:%M"')
        _, result = run('uptime')
        rest = [x.strip() for x in result.split('up', 1)][1]
        parts = [x.strip() for x in rest.split(',')]
        uptime = parts[0] if not 'days' in parts[0] else '%s, %s' % \
            tuple(parts[:2])
        yield panel('System', 'Date: %s<br />Uptime: %s' %
                   (time, uptime), level='info')

    def general(self, request):
        alerts = []
        if needs_restart():
            alerts.append({'message': 'The machine needs to reboot.',
                           'type': 'error'})
        return render('/sys/general', alerts=alerts, updates=get_updates())

    def update(self, request):
        run('apt-get update')
        return self.redirect('/sys')

    def dist_upgrade(self, request):
        if get_updates():
            return Response(app_iter=Updater())
        else:
            alerts = [{'message': 'Software is up to date!'}]
            return render('/sys/general', alerts=alerts)

    def reboot(self, request):
        if 'now' in request.params:
            run('reboot')
        else:
            timer = Timer(1, run, args=('reboot',))
            timer.start()
        alerts = [{'type': 'warn', 'message': 'Rebooting System...'}]
        return render('/sys/general', alerts=alerts)


app = SystemApp()
