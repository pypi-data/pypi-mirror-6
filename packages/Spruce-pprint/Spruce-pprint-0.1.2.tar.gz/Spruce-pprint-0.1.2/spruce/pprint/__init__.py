"""String formatting tools.

This is an extension of :mod:`pprint`.

These tools beautify text strings, especially string representations of
Python objects.  They are not responsible for constructing the original
strings or string representations---that is the prerogative of methods
such as :meth:`~object.__repr__` and :meth:`~object.__str__` and of
application-specific formatting functions.

"""

__copyright__ = "Copyright (C) 2014 Ivan D Vasin"
__credits__ = ["Ivan D Vasin"]
__maintainer__ = "Ivan D Vasin"
__email__ = "nisavid@gmail.com"
__docformat__ = "restructuredtext"

from ._extensions import *
from ._misc import *
from ._pprinters import *
