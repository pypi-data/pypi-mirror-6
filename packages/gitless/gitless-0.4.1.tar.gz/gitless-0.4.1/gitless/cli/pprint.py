# Gitless - a version control system built on top of Git.
# Copyright (c) 2013  Santiago Perez De Rosso.
# Licensed under GNU GPL, version 2.

"""Module for pretty printing Gitless output."""


import sys


# Stdout.

def blank(p=sys.stdout.write):
  return p('#\n')


def msg(msg, p=sys.stdout.write):
  return p('# %s\n' % msg)


def exp(msg, p=sys.stdout.write):
  return p('#   (%s)\n' % msg)


def item(i, opt_msg='', p=sys.stdout.write):
  return p('#     %s%s\n' % (i, opt_msg))


def sep(p=sys.stdout.write):
  return p(
      '########################################################################'
      '########\n')


# Err.

def err(msg, p=sys.stderr.write):
  return p('# %s\n' % msg)


def err_exp(msg, p=sys.stderr.write):
  return p('#   (%s)\n' % msg)


def err_blank(p=sys.stderr.write):
  return p('#\n')


def err_item(i, opt_msg='', p=sys.stderr.write):
  return p('#     %s%s\n' % (i, opt_msg))


# Misc.

def conf_dialog(msg):
  """Gets confirmation from the user.

  Prints a confirmation message to stdout with the given msg and waits for
  user confirmation.

  Args:
    msg: the msg to include in the confirmation.

  Returns:
    True if the user confirmed she wanted to continue or False if otherwise.
  """
  sys.stdout.write('# %s. Do you wish to continue? (y/N)' % msg)
  user_input = raw_input(' ')
  return user_input and user_input[0].lower() == 'y'
