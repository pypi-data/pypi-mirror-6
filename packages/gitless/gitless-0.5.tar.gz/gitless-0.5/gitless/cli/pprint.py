# Gitless - a version control system built on top of Git.
# Copyright (c) 2013  Santiago Perez De Rosso.
# Licensed under GNU GPL, version 2.

"""Module for pretty printing Gitless output."""


import sys


# Stdout.

def blank(p=sys.stdout.write):
  return p('#\n')


def msg(text, p=sys.stdout.write):
  return p('# %s\n' % text)


def exp(text, p=sys.stdout.write):
  return p('#   (%s)\n' % text)


def item(i, opt_text='', p=sys.stdout.write):
  return p('#     %s%s\n' % (i, opt_text))


def sep(p=sys.stdout.write):
  return p(
      '########################################################################'
      '########\n')


# Err.

def err(text, p=sys.stderr.write):
  return p('# %s\n' % text)


def err_exp(text, p=sys.stderr.write):
  return p('#   (%s)\n' % text)


def err_blank(p=sys.stderr.write):
  return p('#\n')


def err_item(i, opt_text='', p=sys.stderr.write):
  return p('#     %s%s\n' % (i, opt_text))


# Misc.

def conf_dialog(text):
  """Gets confirmation from the user.

  Prints a confirmation message to stdout with the given text and waits for
  user confirmation.

  Args:
    text: the text to include in the confirmation.

  Returns:
    True if the user confirmed she wanted to continue or False if otherwise.
  """
  sys.stdout.write('# %s. Do you wish to continue? (y/N)' % text)
  # Python 2/3 compatibility.
  try:
    # Disable pylint's redefined-builtin warning and undefined-variable
    # (raw_input is undefined in python 3) error.
    # pylint: disable=W0622
    # pylint: disable=E0602
    input = raw_input
  except NameError:
    pass
  user_input = input(' ')
  return user_input and user_input[0].lower() == 'y'


def dir_err_exp(fp, subcmd):
  """Prints the dir error exp to stderr."""
  err('{0} is a directory. Can\'t {1} a directory'.format(fp, subcmd))
