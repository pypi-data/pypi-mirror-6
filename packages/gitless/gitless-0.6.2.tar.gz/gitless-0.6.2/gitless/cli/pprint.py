# Gitless - a version control system built on top of Git.
# Licensed under GNU GPL v2.

"""Module for pretty printing Gitless output."""


from clint.textui import puts

import re
import sys

from gitless.core import file as file_lib


SEP = (
    '##########################################################################'
    '######')

# Stdout.

def blank(p=sys.stdout.write):
  puts('#', stream=p)


def msg(text, p=sys.stdout.write):
  puts('# {0}'.format(text), stream=p)


def exp(text, p=sys.stdout.write):
  puts('#   ({0})'.format(text), stream=p)


def item(i, opt_text='', p=sys.stdout.write):
  puts('#     {0}{1}'.format(i, opt_text), stream=p)


def sep(p=sys.stdout.write):
  puts(SEP, stream=p)


# Err.

def err(text):
  msg(text, p=sys.stderr.write)


def err_exp(text):
  exp(text, p=sys.stderr.write)


def err_blank():
  blank(p=sys.stderr.write)


def err_item(i, opt_text='', p=sys.stderr.write):
  item(i, opt_text, p=sys.stderr.write)


# Misc.

def conf_dialog(text):
  """Gets confirmation from the user.

  Prints a confirmation message to stdout with the given text and waits for
  user confirmation.

  Args:
    text: the text to include in the confirmation.

  Returns:
    True if the user confirmed she wanted to continue or False if otherwise.
  """
  msg('{0}. Do you wish to continue? (y/N)'.format(text))
  user_input = get_user_input()
  return user_input and user_input[0].lower() == 'y'


def get_user_input(text='> '):
  """Python 2/3 compatible way of getting user input."""
  global input
  try:
    # Disable pylint's redefined-builtin warning and undefined-variable
    # (raw_input is undefined in python 3) error.
    # pylint: disable=W0622
    # pylint: disable=E0602
    input = raw_input
  except NameError:
    pass
  return input(text)


def dir_err_exp(fp, subcmd):
  """Prints the dir error exp to stderr."""
  err('{0} is a directory. Can\'t {1} a directory'.format(fp, subcmd))


def diff(processed_diff, max_line_digits, p=sys.stdout.write):
  """Uses line-by-line diff information to format lines nicely.

  Args:
    processed_diff: a list of LineData objects.
    max_line_digits: largest number of digits in a line number (for padding).
    p: a writer function (defaults to sys.stdout.write).

  Returns:
    a list of strings making up the formatted diff output.
  """

  def is_unchanged(status):
    """Check if a diff status code does not correspond to + or -.

    Args:
      status: status code of a line.

    Returns:
      True if status is file_lib.DIFF_SAME or file_lib.DIFF_INFO.
    """
    return status == file_lib.DIFF_SAME or status == file_lib.DIFF_INFO

  processed = []
  for index, line_data in enumerate(processed_diff):
    # check if line is a single line diff (do diff within line if so).
    # condition: The current line was ADDED to the file AND
    # the line after is non-existent or unchanged AND
    # the line before was removed from the file AND
    # the line two before is non-existent or unchanged.
    # In other words: bold if only one line was changed in this area.
    if (line_data.status == file_lib.DIFF_ADDED and
       (index == len(processed_diff) - 1 or
           is_unchanged(processed_diff[index + 1].status)) and
       (index - 1 >= 0 and
           processed_diff[index - 1].status == file_lib.DIFF_MINUS) and
       (index - 2 < 0 or is_unchanged(processed_diff[index - 2].status))):
      interest = _highlight(
          processed_diff[index - 1].line[1:], line_data.line[1:])
      if interest:
        # show changed line with bolded diff in both red and green.
        starts, ends = interest
        # first bold negative diff.
        processed[-1] = _format_line(
            processed_diff[index - 1], max_line_digits,
            bold_delim=(starts[0], ends[0]))
        processed += [_format_line(
            line_data, max_line_digits, bold_delim=(starts[1], ends[1]))]
      else:
        processed += [_format_line(line_data, max_line_digits)]
    else:
      processed += [_format_line(line_data, max_line_digits)]
  # TODO: print as we process.
  return p('\n'.join(processed) + '\n')


def _format_line(line_data, max_line_digits, bold_delim=None):
  """Format a standard diff line.

  Args:
    line_data: a namedtuple with the line info to be formatted.
    max_line_digits: maximum number of digits in a line number (for padding).
    bold_delim: optional arg indicate where to start/end bolding.

  Returns:
    a colored version of the diff line using ANSI control characters.
  """
  # Color constants.
  GREEN = '\033[32m'
  GREEN_BOLD = '\033[1;32m'
  RED = '\033[31m'
  RED_BOLD = '\033[1;31m'
  CLEAR = '\033[0m'

  line = line_data.line
  formatted = ''

  if line_data.status == file_lib.DIFF_SAME:
    formatted = (
        str(line_data.old_line_number).ljust(max_line_digits) +
        str(line_data.new_line_number).ljust(max_line_digits) + line)
  elif line_data.status == file_lib.DIFF_ADDED:
    formatted = (
        ' ' * max_line_digits + GREEN +
        str(line_data.new_line_number).ljust(max_line_digits))
    if not bold_delim:
      formatted += line
    else:
      bold_start, bold_end = bold_delim
      formatted += (
          line[:bold_start] + GREEN_BOLD + line[bold_start:bold_end] + CLEAR +
          GREEN + line[bold_end:])
  elif line_data.status == file_lib.DIFF_MINUS:
    formatted = (
        RED + str(line_data.old_line_number).ljust(max_line_digits) +
        ' ' * max_line_digits)
    if not bold_delim:
      formatted += line
    else:
      bold_start, bold_end = bold_delim
      formatted += (
          line[:bold_start] + RED_BOLD + line[bold_start:bold_end] +
          CLEAR + RED + line[bold_end:])
  elif line_data.status == file_lib.DIFF_INFO:
    formatted = CLEAR + '\n' + line

  return formatted + CLEAR


def _highlight(line1, line2):
  """Returns the sections that should be bolded in the given lines.

  Args:
    line1: a line from a diff output without the first status character.
    line2: see line1

  Returns:
    two tuples. The first tuple indicates the starts of where to bold
    and the second tuple indicated the ends.
   """
  start1 = start2 = 0
  match = re.search(r'\S', line1)  # ignore leading whitespace.
  if match:
    start1 = match.start()
  match = re.search(r'\S', line2)
  if match:
    start2 = match.start()
  length = min(len(line1), len(line2)) - 1
  bold_start1 = start1
  bold_start2 = start2
  while (bold_start1 <= length and bold_start2 <= length and
         line1[bold_start1] == line2[bold_start2]):
    bold_start1 += 1
    bold_start2 += 1
  match = re.search(r'\s*$', line1)  # ignore trailing whitespace.
  bold_end1 = match.start() - 1
  match = re.search(r'\s*$', line2)
  bold_end2 = match.start() - 1
  while (bold_end1 >= bold_start1 and bold_end2 >= bold_start2 and
         line1[bold_end1] == line2[bold_end2]):
    bold_end1 -= 1
    bold_end2 -= 1
  if bold_start1 - start1 > 0 or len(line1) - 1 - bold_end1 > 0:
    return (bold_start1 + 1, bold_start2 + 1), (bold_end1 + 2, bold_end2 + 2)
  return None
