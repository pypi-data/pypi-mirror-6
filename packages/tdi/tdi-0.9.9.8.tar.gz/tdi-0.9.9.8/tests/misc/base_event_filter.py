#!/usr/bin/env python
import pprint as _pprint
import sys as _sys
import warnings as _warnings
_warnings.resetwarnings()
_warnings.filterwarnings('error')

from tdi import filters as _filters

class Collector(object):
    def __init__(self):
        self.events = []
        self.canary = 'bla'
    def __getattr__(self, name):
        if name.startswith('handle_'):
            def method(*args):
                self.events.append((name, args))
            return method
        raise AttributeError(name)


class Foo(_filters.BaseEventFilter):
    def __init__(self, builder):
        super(Foo, self).__init__(builder)

    def finalize(self):
        self.builder.handle_blob(self.canary)


class Bar(_filters.BaseEventFilter):
    def __init__(self, builder):
        super(Bar, self).__init__(builder)

    def __getattr__(self, name):
        if name == 'canary':
            raise AttributeError('lol')
        return getattr(self.builder, name)

    def finalize(self):
        self.builder.handle_blob(self.canary)


def test(cls, method, *inp):
    def inner(*expected):
        c = Collector()
        f = cls(c)
        try:
            getattr(f, method)(*inp)
        except (SystemExit, KeyboardInterrupt):
            raise
        except:
            e = _sys.exc_info()[:2]
            c.events.append((e[0].__name__, str(e[1])))
        if c.events == list(expected):
            print "OK", method, inp
        else:
            print "FAIL", method, inp
            _pprint.pprint(tuple(c.events))
    return inner

#_test, test = test, lambda *a, **k: lambda *e: None


test(Foo, 'foo', 'bar', 'baz')(('AttributeError', 'foo'),)
test(Foo, 'handle_blub', 'plop', 'poeh')(('handle_blub', ('plop', 'poeh')),)
test(Foo, 'finalize')(('handle_blob', ('bla',)),)

test(Bar, 'foo', 'bar', 'baz')(('AttributeError', 'foo'),)
test(Bar, 'handle_blub', 'plop', 'poeh')(('handle_blub', ('plop', 'poeh')),)
test(Bar, 'finalize')(('AttributeError', 'lol'),)

