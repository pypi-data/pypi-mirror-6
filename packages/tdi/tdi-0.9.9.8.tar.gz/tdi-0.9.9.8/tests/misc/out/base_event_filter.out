OK foo ('bar', 'baz')
OK handle_blub ('plop', 'poeh')
OK finalize ()
OK foo ('bar', 'baz')
OK handle_blub ('plop', 'poeh')
OK finalize ()
